src/layers/core/grid-layer/grid-layer.js

```diff
@@ -54,25 +56,62 @@ function _needsReProjectPoints(oldProps, props) {
   return oldProps.cellSize !== props.cellSize;
 }
 
+function _percentileChanged(oldProps, props) {
+  return oldProps.lowerPercentile !== props.lowerPercentile ||
+    oldProps.upperPercentile !== props.upperPercentile;
+}
+
+function _needsReSortBins(oldProps, props) {
+  return oldProps.getColorValue !== props.getColorValue;
```

Comment:
I would recommend giving this update criterion another round of thought, as it is inconsistent with updateTriggers which we use to make sure we don't update when functions change.

This is to handle the common cases of ad-hoc functions and bound functions:
```
new Layer({
   getColorValue: () => {}, // new function every render,
   getColorValue: this._getColorValue.bind(this), // new function every render,
});
```